src/app/core/standalone/component/suggest/search-box/search-box.component.ts

```diff
@@ -0,0 +1,272 @@
+  private blurIfOutside(targetElement: undefined): void {
+    const clickedOrFocusedInside = this.searchBox.nativeElement.contains(targetElement);
+    if (!clickedOrFocusedInside) {
+      this.blur();
+    }
+  }
+
+  // simple blur method to remove focus from search input
+  private blur() {
+    this.handleFocus(false);
+    this.searchInput.nativeElement.blur();
+  }
+
+  // handle focus status of search box
+  handleFocus(scaleUp: boolean) {
+    this.updateMobileSuggestLayerHeight();
+
+    if (scaleUp) {
+      this.searchBoxFocus = true;
+      // this.searchBoxScaledUp is set using transitionend event
+    } else {
+      this.searchBoxFocus = false;
+      this.searchBoxScaledUp = false;
+    }
+  }
+
+  // manually set focus on search input
+  // the explicit function call in the component is needed to get the focus working in iOS devices
+  setFocusOnSearchInput() {
+    this.searchInput.nativeElement.focus();
+  }
+
+  // handle the user input
+  handleInput(source: EventTarget) {
+    const inputValue = (source as HTMLInputElement).value;
+    this.inputSearchTerms$.next(inputValue);
+    if (inputValue === '') {
+      // clear suggestions in state when input is set to empty
+      this.shoppingFacade.clearSuggestSearchSuggestions();
+    }
+  }
+
+  // reset all and blur the input
+  resetInput() {
+    this.blur();
+    this.inputSearchTerms$.next('');
+    this.shoppingFacade.clearSuggestSearchSuggestions();
+  }
+
+  // handle the reset button
+  handleResetButton(event: Event) {
+    event.stopPropagation(); // important to prevent any other event listeners from firing
+    this.inputSearchTerms$.next('');
+    this.shoppingFacade.clearSuggestSearchSuggestions();
+  }
+
+  // submit the search form
+  submitSearch(suggestedTerm: string) {
+    if (!suggestedTerm) {
+      this.setFocusOnSearchInput();
+      return false;
+    }
+
+    // add the suggested term to the input field
+    this.inputSearchTerms$.next(suggestedTerm);
+
+    this.router.navigate(['/search', suggestedTerm]);
+    this.blur();
+    return false; // prevent form submission
+  }
+
+  // getter method to check if the input search term has more than 3 characters
+  get hasMinimumCharCount(): boolean {
+    let term = '';
+    this.inputSearchTerms$.pipe(takeUntilDestroyed(this.destroyRef)).subscribe(value => (term = value));
+    return term.length > 2;
+  }
+
+  hasSearchedTerms(): boolean {
+    let terms = [];
+    this.shoppingFacade.recentSearchTerms$.pipe(takeUntilDestroyed(this.destroyRef)).subscribe(data => (terms = data));
+    return terms.length > 0;
+  }
+
+  // set CSS variable for suggest layer height on mobile devices to prevent keyboard overlay issues
+  private updateMobileSuggestLayerHeight = () => {
+    if (!SSR && this.deviceType === 'mobile') {
+      clearTimeout(this.resizeTimeout);
+
+      // timeout to wait for keyboard animation to finish
+      this.resizeTimeout = setTimeout(() => {
```

Comment:
It will cause issues if a timer is set before the angular application gets stable, that's why you have to wait for the appRef.isstable before you can trigger the timer
In addition: is it really necessary to have such a complicated mobile support here? It seems to me that this component should be better simplified and projects should add extra functionality if needed